parachain/pallets/ethereum-light-client/src/lib.rs

```diff
@@ -265,6 +265,48 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Reset's the internal state of the light client to a previously imported header.
+		///
+		/// `forked_at` - The hash of the header where the fork originated. The common ancestor.
+		///
+		/// This is only meant to be used in staging environments where long range forks
+		/// can happen.
+		///
+		/// Requires sudo user.
+		#[pallet::weight(1_000_000)]
+		#[transactional]
+		pub fn force_reset_to_fork(origin: OriginFor<T>, forked_at: H256) -> DispatchResult {
+			ensure_root(origin)?;
+
+			let stored_header = <Headers<T>>::get(forked_at).ok_or(Error::<T>::MissingHeader)?;
+
+			let required_descendants = T::DescendantsUntilFinalized::get() as usize;
+			let best_block_id =
+				EthereumHeaderId { number: stored_header.header.number, hash: forked_at };
+
+			let current_finalized = <FinalizedBlock<T>>::get();
+			if best_block_id.number >= current_finalized.number {
+				return Err("Cannot reset to fork after the current finalized block.".into())
+			}
+
+			let (new_finalized_hash, _) =
+				Self::find_finalized_ancestor(forked_at, required_descendants).ok_or(
+					"Cannot reset to fork if it does not have the required number of decendants.",
+				)?;
+
+			let finalized_stored_header =
```

Comment:
This retrieval is redundant, since `find_finalized_ancestor` already returns an `EthereumHeader`, which you're ignoring. And you can get the block number from the `EthereumHeader`, so need to retrieve the StoredHeader.